indexer-opts/src/lib.rs

```diff
@@ -0,0 +1,245 @@
+VALUES ($1, $2, now(), $3, $3, $4)
+ON CONFLICT (indexer_id) DO UPDATE
+    SET start_block_height = EXCLUDED.start_block_height,
+        last_processed_block_height = EXCLUDED.start_block_height,
+        end_block_height = EXCLUDED.end_block_height,
+        indexer_started_at = now()
+    WHERE __meta.indexer_id = EXCLUDED.indexer_id
+                "#,
+                indexer_id,
+                indexer_type,
+                block_height,
+                end_block_height,
+            )
+            .execute(db_with_meta_data_pool)
+            .await?;
+            Ok(())
+        }
+    }
+}
+
+impl Opts {
+    // returns a Lake Config object where AWS credentials are sourced from .env file first, and then from .aws/credentials if not found.
+    // https://docs.aws.amazon.com/sdk-for-rust/latest/dg/credentials.html
+    pub async fn to_lake_config(
+        &'static self,
+        db_with_meta_data_pool: &sqlx::Pool<sqlx::Postgres>,
+    ) -> anyhow::Result<near_lake_framework::LakeConfig> {
+        let config_builder = near_lake_framework::LakeConfigBuilder::default();
+
+        tracing::info!("CHAIN_ID: {:?}", self.chain_id);
+
+        let start_block_height = match self.start_mode {
+            StartMode::FromLatest => {
+                latest_block_height(self.rpc_url.as_ref().unwrap_or_else(|| {
+                    panic!("`rpc-url` must be provided for `--start-mode from-lastest")
+                }))
+                .await
+            }
+            StartMode::FromInterruption => {
+                match last_processed_block_height(&self.indexer_id, db_with_meta_data_pool).await {
+                    Ok(last_processed_block_height) => last_processed_block_height,
+                    Err(err) => {
+                        tracing::warn!(
+                            "Failed to fetch `last_processed_block_height` from meta data. Falling back to provided `start_block_height`\n{:#?}",
+                            err,
+                        );
+                        self.start_block_height.unwrap_or_else(|| panic!("`__meta` for INDEXER ID {} doesn't exist `start-from-block-height` must be provided", self.indexer_id))
+                    }
+                }
+            }
+        };
+
+        Ok(match self.chain_id {
+            ChainId::Mainnet => config_builder.mainnet(),
+            ChainId::Testnet => config_builder.testnet(),
+        }
+        .start_block_height(start_block_height)
+        .build()?)
+    }
+}
+
+async fn latest_block_height(rpc_url: &str) -> u64 {
+    let client = JsonRpcClient::connect(rpc_url);
+    let request = methods::block::RpcBlockRequest {
+        block_reference: BlockReference::Finality(Finality::Final),
+    };
+
+    let latest_block = client
+        .call(request)
+        .await
+        .unwrap_or_else(|_| panic!("Failed to fetch final block from RPC {}", rpc_url));
+
+    latest_block.header.height
+}
+
+async fn last_processed_block_height(
```

Comment:
can we make all the functions staring with the verb?
`get` here

I usually don't have any opinion about it, but here the name is the same as the variable and it makes me crazy, we invoke the variable